src/ota/ota_handler.rs

```diff
@@ -314,4 +307,171 @@ mod tests {
             OTAStatus::InProgress.to_status_code()
         );
     }
+
+    use crate::astarte::MockPublisher;
+    use crate::ota::rauc::BundleInfo;
+    use crate::ota::MockOTA;
+    use crate::repository::MockStateRepository;
+
+    fn get_string_value_from_update_error(error: DeviceManagerError) -> String {
+        match error {
+            DeviceManagerError::UpdateError(value) => value,
+            _ => "".to_owned(),
+        }
+    }
+
+    #[test]
+    fn handle_ota_event_bundle_not_compatible() {
+        let mut publisher = MockPublisher::new();
+        publisher
+            .expect_send_object()
+            .returning(|_, _: &str, _: OTAResponse| Ok(()));
+
+        let mut ota = MockOTA::new();
+        ota.expect_info().returning(|_: &str| {
+            Ok(BundleInfo {
+                compatible: "rauc-demo-x86".to_string(),
+                version: "1".to_string(),
+            })
+        });
+
+        ota.expect_compatible()
+            .returning(|| Ok("rauc-demo-arm".to_string()));
+
+        let mut state_mock = MockStateRepository::<PersistentState>::new();
+        state_mock.expect_write().returning(|_| Ok(()));
+
+        let mut ota_handler = OTAHandler {
+            ota: Box::new(ota),
+            state_repository: Box::new(state_mock),
+            download_file_path: "".to_owned(),
+        };
+
+        let result = block_on(ota_handler.handle_ota_event(&publisher, "", ""));
+        assert!(result.is_err());
+
+        let result_error = get_string_value_from_update_error(result.err().unwrap());
+        let error_expected = "bundle is not compatible";
+        assert_eq!(error_expected, result_error);
```

Comment:
Could this be made clearer by explicitly letting `get_string_value_from_update_error` fail if `result.err().unwrap()` is not an `UpdateError`?